src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -186,6 +183,38 @@ static bool fifo_alloc(uint8_t rhport, uint8_t ep_addr, uint16_t packet_size) {
   return true;
 }
 
+static inline bool dma_enabled(uint8_t rhport)
+{
+  // DMA doesn't support fifo transfer
+#ifdef TUD_AUDIO_PREFER_RING_BUFFER
```

Comment:
Don't want to depend on class driver but maybe it's the simplest way.

Reply:
CFG_TUD_DWC2_DMA must be enabled at compile time, audio driver/user should adapt accordingly.